### PR TITLE
Add actions validation for unrelated blocks

### PR DESCRIPTION
**Pull request by ajmaddaford**
### PR Context

Adds validation for the required actions (`RemoveUnrelatedRelationships` and `AddUnrelatedRelationships`) on unrelated blocks.

### Checklist

* [ ] eq-translations updated to support any new schema keys which need translation

### CODE REVIEW

File: app/validators/blocks/relationship_collector_validator.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from app import error_messages
 from app.validators.blocks.block_validator import BlockValidator
+from app.validators.blocks.unrelated_block_validator import UnrelatedBlockValidator
```
Comment (JamesGardiner) [inline app/validators/blocks/relationship_collector_validator.py:3]: A minor niggle, but these imports could be:

```
from app.validators import BlockValidator
from app.validators import UnrelatedBlockValidator
```

with some small changes to the init modules

**Review by JamesGardiner — APPROVED** [review]
LGTM - minor suggestion to improve readability of imports, but it's not blocking

File: app/validators/blocks/relationship_collector_validator.py (hunk 1 above)
Comment (ajmaddaford) [reply app/validators/blocks/relationship_collector_validator.py:3]: I agree, but this is consistent with the imports throughout validator. This change should be made across all files in a separate pull request.

File: app/validators/blocks/unrelated_block_validator.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    def validate_answer_actions(self):
+        expected_actions = ["AddUnrelatedRelationships", "RemoveUnrelatedRelationships"]
+        questions = self.questionnaire_schema.get_all_questions_for_block(self.block)
+        for question in questions:
+            actions = []
+            for answer in question["answers"]:
+                for option in answer["options"]:
+                    if "action" in option:
+                        actions.append(option["action"]["type"])
```
Comment (MebinAbraham) [inline app/validators/blocks/unrelated_block_validator.py:20]: Ideal for a comprehension?
```suggestion
            actions = [
                option["action"]["type"]
                for answer in question["answers"]
                for option in answer["options"]
                if "action" in option
            ]
```

File: app/validators/blocks/unrelated_block_validator.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+            if sorted(actions) != expected_actions:
```
[the remark this answers is not in the document]
Comment (ajmaddaford) [reply app/validators/blocks/unrelated_block_validator.py:22]: Yes, that is how the unrelated block works.

[the remark this answers is not in the document]
Comment (MebinAbraham) [reply app/validators/blocks/unrelated_block_validator.py:22]: Ya I just mean from runner's point, it doesn't care as long as the action exists. I.e if you had multiple options for yes/no like you have with driving collectors. Obv this requirement shouldn't chnage so not an issue.

File: app/validators/blocks/unrelated_block_validator.py (hunk 2 above)
Comment (ajmaddaford) [reply app/validators/blocks/unrelated_block_validator.py:20]: 👍

**Review by MebinAbraham — APPROVED** [review]

File: app/validators/blocks/unrelated_block_validator.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ajmaddaford) [reply app/validators/blocks/unrelated_block_validator.py:23]: I think validation gets unnecessarily more complicated, to allow for scenarios we have no use cases for.

**Pull request merged**